sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -832,7 +832,8 @@ private[hive] class HiveClientImpl(
     val (partCols, schema) = table.schema.map(toHiveColumn).partition { c =>
       table.partitionColumnNames.contains(c.getName)
     }
-    if (schema.isEmpty) {
+    if (schema.isEmpty&& table.properties.getOrElse(
+      HiveExternalCatalog.DATASOURCE_SCHEMA_NUMPARTS, "0").toInt != 0) {
```

Comment:
why we need this condition? I think we store schema in table properties for all kinds of tables.